### DISCUSSION TITLE
Operation Completed Successfully

### DISCUSSION THREAD

**Discussion by X-Jaber** [neos-sdi/adfsmfa#338] [General]
We have been facing this issue for a while now, unfortunately there weren’t much of resources online to help us resolve this issue,
I was wondering if it is familiar for you and it something you could help us with. 
 
We found it was mentioned here [issue #198](https://github.com/neos-sdi/adfsmfa/issues/198)
we could see it has been resolved in 3.1.2202.2 based on the [change log](https://github.com/neos-sdi/adfsmfa/wiki/12-ChangeLog) but yet we are on a later release 3.1.2207.0v. and facing the same issue was this issue addressed in newer versions?

what happens?
when our users’ login enters the correct OTP code they get "Operation Completed Successfully " attached for reference, 
and they have to enter the OTP multiple times until it works.

We have made a lot of restarts but we are not quite sure what is wrong and where to start troubleshooting this issue

![image](https://github.com/neos-sdi/adfsmfa/assets/167176976/68030124-efbe-469a-8ea1-31c6d8215278)

**Comment by redhook62** [comment]
Hi @X-Jaber 

I validated your configuration and it is correct.

I have no precise idea of your problem, because I have never been able to notice it.

This type of error comes from a Win32 call, we have very few in our code. 
this can come from your operating system. 
These Win32 calls are in no way used during the authentication process you describe.

The easiest thing is for you to test the latest version, it won't be any worse.
Can you give us feedback on the latest version ?

I am moving this issue to discussion, because the people who had this problem may also be able to help you.

regards

**Comment by derSchweiger** [comment]
Hi,

now and then we experience the same issue. We have already installed the latest version.
@X-Jaber do you have multiple ADFS servers in your farm or only a single instance? Can you see any error events in the eventlog?

**Comment by behrouzamiri** [comment]
Experienced the same issue.
I'm following this behaviour, and it seems there's something in deployments that are involving multiple nodes, being load balanced behind a reverse proxy. Whoever experienced such an issue had multiple nodes behind a reverse proxy!

**Comment by Iddunx** [comment]
Hi,
We're experiencing a similar issue. We have an ADFS cluster with two nodes (no load balancing), and MFA is configured using an MS SQL database.

The issue tends to occur around the fifth day of the MFA service running. Restarting the MFA service temporarily resolves the problem.

We’d like to enable more detailed debug logging, especially related to TOTP code handling. Is there a way to capture what’s happening within the MFA service when users click "Sign in" and encounter the message: “The operation completed successfully”?

Our configuration includes a log path setting:
```
LogPath=C:\Logs\MFA.log
````
However, this log only seems to capture OTP-related actions.


@redhook62, could you please advise on enabling more detailed logs or any troubleshooting steps?
